frontend/src/pages/viewtraining/viewtraining.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect, useContext } from "react";
+import { Button, Typography, Divider, Box, FormControl, Select, MenuItem, Grid, TextField } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+
+import { AuthContext } from "../../context/auth";
+import api from "../../helpers/api";
+
+const useStyles = makeStyles({
+    bold: {
+        fontWeight: 600
+    },
+    underline: {
+        textDecorationLine: 'underline'
+    },
+    italic: {
+        fontStyle: 'italic'
+    },
+    formControl: {
+        minWidth: 200,
+    },
+})
+
+
+export default function ViewCourse(props) {
+    const courseId = props.match.params.courseId;
+    const classes = useStyles();
+    const { authState } = useContext(AuthContext);
+    const [courseState, setCourseState] = useState(undefined);
+    const [editState, setEditState] = useState(true);
+
+    const handleChange = (e) => {
+        const { id, value } = e.target;
+        setCourseState({ ...courseState, [id]: value })
+    }
+
+    const handleDropdown = (e) => {
+        const { value } = e.target;
+        setCourseState({ ...courseState, type: value })
+    }
+
+    const handleEdit = async (e) => {
+        if (!editState) {
+            console.log(courseState)
+            const res = await api.course.update(authState.token, courseId, courseState);
+        }
+        setEditState(!editState);
+    }
+
+    const fetchData = async () => {
+        const res = await api.course.get(authState.token, courseId);
```

Comment:
You need to create this function in `helpers/api/course.js` via the `helpers/api/index.js`